docs/rules/index.md

```diff
@@ -4,6 +4,14 @@ This plugin provides the following rules.
 
 - 🔧 mark means that the `--fix` option on the [command line](https://eslint.org/docs/user-guide/command-line-interface#fixing-problems) can automatically fix some of the problems reported by the rule.
 
+## ES2025
+
+There is a config that enables the rules in this category: [`no-new-in-esnext`]
+
+| Rule ID | Description |    |
+|:--------|:------------|:--:|
+| [es-x/no-set-operation-methods](./no-set-operation-methods.md) | disallow methods like union and intersection of Set. |  |
```

Comment:
The current approach confuses me. Usually, I use `eslint-plugin-es(-x)` to forbid specific methods / features, not specific ES versions / proposals. If some methods from the proposal are available, it does not mean that all of them are available. For example, for this proposal, early Safari / Bun implementations add methods from this proposal... but missed `Set.prototype.difference`, https://github.com/oven-sh/bun/issues/2309. After that, when the proposal was already available in some engines, the algorithm of `Set.prototype.intersection` was changed. Etc. I think that it could be good to have a rule for the proposal and rules for each method / feature from it.

Reply:
Thank you for your comment!
Thanks to the information you provided, I think it would be good to provide this rule separately for each method :+1:

However, implementing rules for each proposal has been done before.
https://github.com/mysticatea/eslint-plugin-es/blob/master/docs/rules/no-string-prototype-trimstart-trimend.md
Therefore, I'm not sure whether we should provide separate rules for all proposals in the future by methods / features. I think maybe we need to judge individually.

Or do you think we should also split the already implemented rules?

Reply:
> However, implementing rules for each proposal has been done before.

Yes, I know. I think that it could be good to provide separate versions of rules even for not recent features.

As an option, it could be options for proposals rules - however, it's not so intuitive as separate rules with names of methods.

Reply:
> I meant to leave existing rules for proposals and add separate rules for each feature - both ways make sense for different approaches. Like in core-js - core-js/proposals/set-methods vs core-js/actual/set/difference entries.

Ah, I see 😄 
However, I think it would be difficult to create a rule that combines multiple rules.
In order to do that, I think it is necessary to separate all the rules of the plugin by method / feature and provide a shareable config for each proposal.
But right now, I'm not sure if it's useful for people. If you think it would be useful to people, could you please open a new issue? :pray:

Reply:
I don't need it for my cases, however, I think that it could be a useful improvement, so let's open an issue.